### PR TITLE
WIP pull request - ERC viewable as galley 

### PR DESCRIPTION
**Pull request by tnier01**
@nuest 
Here my first steps with the plugin!

### CODE REVIEW

**Comment by tnier01** [comment]
@nuest @Fmazin 
Here a first WIP pull request. It would be great if you @Fmazin take a look at whether everything is understandable and works as presented last week and if necessary comment or accept.

File: build/config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const config= {
+    "baseUrl": "https://o2r.uni-muenster.de/api/v1/",
+    "ercUrl": "http://localhost/erc/",
+    "title":" | o2r Demoserver",
```
Comment (nuest) [inline build/config.js:4]: Can we switch the title dynamically to include the journal name? Would be a nice feature, not part of this PR obviously.

File: build/manifest.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "React App",
```
Comment (nuest) [inline build/manifest.json:2]: Change short name and name to something more reasonable.

File: ojsErcPlugin.inc.php [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+
+				preg_match('"\.\/logo\.png"', $readHtmlFile, $logoOld);
+				$logoNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($logoOld[0], 1); 
+				$adaptedLogo = str_replace($logoOld[0], $logoNew, $adaptedMainJs);
+
+				preg_match('"\.\/manifest\.json"', $readHtmlFile, $manifestOld);
+				$manifestNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($manifestOld[0], 1); 
+				$adaptedManifest = str_replace($manifestOld[0], $manifestNew, $adaptedLogo);
+
+				preg_match('"\.\/config\.js"', $readHtmlFile, $configOld);
```
Comment (nuest) [inline ojsErcPlugin.inc.php:123]: Can you encapsulate this in a function? Looks very repetitive.

File: version.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE version SYSTEM "../../../lib/pkp/dtd/pluginVersion.dtd">
```
Comment (nuest) [inline version.xml:2]: This path does not exist in this repository - I assume that is not a problem? Or exchange the dtd URL with a correct online URL?

File: version.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+<version>
+	<application>ojs-erc-plugin</application>
+	<type>plugins.generic</type>
+	<release>0.1.0.0</release>
```
Comment (nuest) [inline version.xml:6]: Are 4 digits required for the version, or can we also stick to `major.minor.bugfix` ?

File: version.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+	<date>2019-05-15</date>
```
Comment (nuest) [inline version.xml:7]: Please start the notes for creating releases and document how this file needs to be updated.

**Review by nuest — COMMENTED** [review]
Are the `...chunk.css/js/map` files not part of the UI download? Do they need to be saved in the plugin?

Regarding my other comments: I'm happy to merge without them being handled, we can create issues instead.

File: build/manifest.json (hunk 2 above)
Comment (tnier01) [reply build/manifest.json:2]: @NJaku01 Can you take care of it so that it is done in general for the build?

File: build/config.js (hunk 1 above)
Comment (tnier01) [reply build/config.js:4]: issue created https://github.com/o2r-project/ojs-erc-plugin/issues/18

File: build/manifest.json (hunk 2 above)
Comment (NJaku01) [reply build/manifest.json:2]: I have adapted it in my fork, the changes will be available in the build with the next release

File: ojsErcPlugin.inc.php (hunk 3 above)
Comment (tnier01) [reply ojsErcPlugin.inc.php]: Yeah is done here: https://github.com/o2r-project/ojs-erc-plugin/pull/1/commits/a23c3354f9655c3bede6e6a2bf77352e7c441df2

File: version.xml (hunk 4 above)
Comment (tnier01) [reply version.xml:2]: This is a default of the ojs repository. In the OJS repository the path leads to the plugin version. So I think its correct.

File: version.xml (hunk 5 above)
Comment (tnier01) [reply version.xml]: Now we can also stick to 3 digits, the corresponding commit can be found here: https://github.com/o2r-project/ojs-erc-plugin/pull/1/commits/eaea77a44ed699ff3321899fe6290f7d33fe0dd7

File: version.xml (hunk 6 above)
Comment (tnier01) [reply version.xml]: The notes are now available via a description in the README.md and the CHANGELOG.md. The corresponding commit can be found here: https://github.com/o2r-project/ojs-erc-plugin/pull/1/commits/eaea77a44ed699ff3321899fe6290f7d33fe0dd7

**Comment by tnier01** [comment]
> Are the `...chunk.css/js/map` files not part of the UI download? Do they need to be saved in the plugin?
> 
> Regarding my other comments: I'm happy to merge without them being handled, we can create issues instead.

Concerning your question, the `...chunk.css/js/map` files are part of the UI download and thus they are downloaded within the build folder. Of course the build folder does not have to be part of this repository, because it is downloaded automatically when the plugin is activated. 
All comments have been edited appropriately or converted to issues and can be resolved if agreed.
